### PR TITLE
Docs: migration documentation for MDX2 in storybook CLI

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybookjs/storybook/issues/20091

## What I did

I changed the auto-migration to link to our docs, (which in turn links to the MDX migration guide)

This way we have a single place to link to regaridng all MDX migration. (which might become a few more still).

### CODE REVIEW

File: docs/writing-docs/mdx.md [hunk 1]
```diff
@@ -281,3 +281,32 @@ You can also use anchors to target a specific section of a page:
...
 - [Publishing docs](./build-documentation.md) to automate the process of publishing your documentation
+
+## upgrading to MDX2
+
+In Storybook 7.0 MDX2 (https://mdxjs.com/blog/v2/) is used, which contains breaking changes from V1.
+
+When you run `npx storybook upgrade` to upgrade your storybook, part of that script will help you upgrade your MDX files to MDX2; or take you part of the way there. Some manual work might be required, depending on your own needs.
+We suggest you do read this MDX2 migration guide here: https://mdxjs.com/migrating/v2/#update-mdx-files
+
+In MDX1 storybook automatically added "Github Flavoured Markdown" support (GFM for short). In Storybook 7.0 this is no longer the case, and therefore if you want this, you need to modify your storybook's `mina.js` config file to add it.
```
Comment (joshwooding) [inline docs/writing-docs/mdx.md:292]:
```suggestion
In MDX1 storybook automatically added "Github Flavoured Markdown" support (GFM for short). In Storybook 7.0 this is no longer the case, and therefore if you want this, you need to modify your storybook's `main.js` config file to add it.
```

**Comment by ndelangen** [comment]
TODO: The recipe should be changed to become either a TS files or a `.mjs` file.

**Comment by jonniebigodes** [comment]
@ndelangen I've just pushed some changes to this pull request, which I hope you don't mind. I've updated the link in the CLI to point at the existing documentation (for the 7.0 docs, we'll need to follow up once Storybook 7.0 finally comes out), which covers your change and removed the upgrading bit as it was redundant.

Let me know if you have questions about this and I'll be more than glad to answer them.

**Comment by ndelangen** [comment]
@shilman I've kind of lost track of what needs to be done here. Is this PR mergable as-is? Or is more work needed here? Does this PR still make sense?

**Pull request merged**